src/wasm/wasm-binary.cpp

```diff
@@ -2897,6 +2878,1370 @@ void WasmBinaryReader::readVars() {
+        }
+        case BinaryConsts::I64AtomicStore16: {
+          auto [mem, align, offset] = getMemarg();
+          return builder.makeAtomicStore(2, offset, Type::i64, mem);
+        }
+        case BinaryConsts::I64AtomicStore32: {
+          auto [mem, align, offset] = getMemarg();
+          return builder.makeAtomicStore(4, offset, Type::i64, mem);
+        }
+        case BinaryConsts::I64AtomicStore: {
+          auto [mem, align, offset] = getMemarg();
+          return builder.makeAtomicStore(8, offset, Type::i64, mem);
+        }
+
+#define RMW(op)                                                                \
+  case BinaryConsts::I32AtomicRMW##op: {                                       \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 4, offset, Type::i32, mem);          \
+  }                                                                            \
+  case BinaryConsts::I32AtomicRMW##op##8U: {                                   \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 1, offset, Type::i32, mem);          \
+  }                                                                            \
+  case BinaryConsts::I32AtomicRMW##op##16U: {                                  \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 2, offset, Type::i32, mem);          \
+  }                                                                            \
+  case BinaryConsts::I64AtomicRMW##op: {                                       \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 8, offset, Type::i64, mem);          \
+  }                                                                            \
+  case BinaryConsts::I64AtomicRMW##op##8U: {                                   \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 1, offset, Type::i64, mem);          \
+  }                                                                            \
+  case BinaryConsts::I64AtomicRMW##op##16U: {                                  \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 2, offset, Type::i64, mem);          \
+  }                                                                            \
+  case BinaryConsts::I64AtomicRMW##op##32U: {                                  \
+    auto [mem, align, offset] = getMemarg();                                   \
+    return builder.makeAtomicRMW(RMW##op, 4, offset, Type::i64, mem);          \
+  }
+
+          RMW(Add);
+          RMW(Sub);
+          RMW(And);
+          RMW(Or);
+          RMW(Xor);
+          RMW(Xchg);
```

Comment:
Indentation looks odd?

Reply:
These are indented more than the previous macro uses because they are inside the nested switch for the atomic prefix.